temporian/core/operators/lag.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2021 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Lag operator."""
+
+from temporian.core import operator_lib
+from temporian.core.data.event import Event
+from temporian.core.data.feature import Feature
+from temporian.core.data.sampling import Sampling
+from temporian.core.operators.base import Operator
+from temporian.proto import core_pb2 as pb
+
+
+class LagOperator(Operator):
+    """Lag operator."""
+
+    def __init__(
+        self,
+        event: Event,
+        duration: float,
+    ):
+        super().__init__()
+
+        # inputs
+        self.add_input("event", event)
+
+        self.add_attribute("duration", duration)
+
+        # outputs
+        output_features = [  # pylint: disable=g-complex-comprehension
+            Feature(
+                name=f"lag_{f.name()}",
```

Comment:
Make the prefix dependent on the sign of the duration.

Reply:
Yes, it's part of the goal. But I don't have the `Duration` class and any `Duration` helper function. I could wait for you to merge the PR of SMA and then rebase to main in order to have the functions. If you think that the SMA PR will take much more to merge, you can open a PR only for `Duration` with its functions and merge that PR to main.

Reply:
I still have the question of why we define each feature name twice. We define them in core and then in the implementation. Any comments on this? @getchebarne @ianspektor